cpu/imxrt117x-cm4/lib/mu.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX RT117x Messaging Unit driver
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include "mu.h"
+#include "interrupt.h"
+#include "cm4.h"
+#include "string.h"
+
+
+enum { btr0 = 0, btr1, btr2, btr3, brr0, brr1, brr2, brr3, bsr, bcr };
+
+
+typedef union {
+	struct {
+		char bno;
+		char payload[3];
+	} __attribute__((packed)) b;
+	uint32_t w;
+} packet_t;
```

Comment:
Yes, lot of code is duplicated between mu.c and imxrt117x-cm4.c. These are both drivers for the same hardware, but code can't be really shared - lib and example directories are here temporaly, I think it would be best if these were placed in separate repository.